packages/vite/src/node/plugins/html.ts

```diff
@@ -297,9 +297,11 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
               someScriptsAreAsync ||= isAsync
               someScriptsAreDefer ||= !isAsync
             } else if (url && !isPublicFile) {
-              config.logger.warn(
-                `<script src="${url}"> in "${publicPath}" can't be bundled without type="module" attribute`
-              )
+              if (!isExcludedUrl(url)) {
```

Comment:
This has been on my to-fix list for a long while. Thanks for fixing this 😄 

My only concern with this is that `isExcludedUrl` also runs `checkPublicFile`, which is already covered by `isPublicFile`. But given this warning don't always happen, I think it's still fine.

Reply:
@bluwy  I thought it was too much to create a new function similar to isExcludedUrl() but without the checkPublicFile() call. Felt more like duplicating code than optimization. However, should you guys want to, I can create a new PR.

Reply:
I think it's fine the way it is now 👍